src/utils.ts

```diff
@@ -791,9 +837,9 @@ export function equals(a, b) {
     var length, i, keys;
```

Comment:
length is defined but never used.

If this function is called on a lot of elements, I advise you at least benchmark the two versions before changing it, as I suspect the `length = a.length;` part was put here for performances purpose.

Evaluating attribute with "." several times can be way more costly than setting it into a variable and saving the evaluating cost while iterating over a large number of element

Reply:
Went back in past